### PR TITLE
Implemented Multiplayer Functionality with WebSockets

### PR DESCRIPTION
**Pull request by EvanW1999**
Add server and client websockets for joining game, updating scores, signaling beginning and end of games, and chat.

### CODE REVIEW

File: client/src/components/game.tsx [hunk 1]
```diff
@@ -1,223 +1,306 @@
...
+  const [question, setQuestions] = useState<string[]>([]);
+
+  console.log(gameInfo);
+
+  useEffect(() => {
+    const getQuestions = async (): Promise<void> => {
+      const res = await fetch(`api/v1/game/${gameInfo.id}/questions`);
+      if (res.ok) {
+        const data = await res.json();
+        setQuestions(data["questions"]);
```
Comment (timothycobrien) [inline client/src/components/game.tsx:92]: Might want to modify later to not load all questions at once

File: client/src/components/game.tsx [hunk 2]
```diff
@@ -1,223 +1,306 @@
...
-        if (event.key === 'Enter') {
-            event.preventDefault();
-            event.stopPropagation();
-            onSubmit();
-        }
+  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>): void => {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      event.stopPropagation();
+      onSubmit();
```
Comment (timothycobrien) [inline client/src/components/game.tsx:208]: Might want to modify this to auto-submit x milliseconds after last keystroke is pressed, pretty clean ui

Comment (timothycobrien) [reply client/src/components/game.tsx:208]: That or enter i mean

File: client/src/components/game.tsx [hunk 3]
```diff
@@ -231,235 +314,299 @@ const AnswerBoxBase = styled.div`
...
-            {message.sender}: {message.text}
-        </div>
-    ));
-    return(<div style={{"overflow": "scroll", "gridArea":"chat", "display":"flex", "flexDirection": "column-reverse"}}>{messageBox}</div>);
+const ChatBox = ({ name, id }: { name: string; id: string }): ReactElement => {
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [myMessage, updateMessage] = useState<string>("");
+  const socket: Socket = useContext(SocketContext);
+
+  useEffect(() => {
```
Comment (timothycobrien) [inline client/src/components/game.tsx:366]: What's the point of this `useEffect`

File: server/src/api/v1/game.py [hunk 4]
```diff
@@ -117,3 +124,68 @@ def join_game():
...
+        emit("update_players", update_game_players(
+            game_id), room=game_id)
+
+    else:
+        emit("validate_answer", False)
+
+
+@socketio.on("send_chat")
+def send_chat(chat_data: Json):
+    game_id: str = chat_data["game_id"]
```
Comment (timothycobrien) [inline server/src/api/v1/game.py:180]: Where does this update the chat itself

**Review by timothycobrien — APPROVED** [review]

File: server/src/api/v1/game.py (hunk 4 above)
Comment (EvanW1999) [reply server/src/api/v1/game.py:180]: It just emits to all the sockets in the room, there's no db for the chat atm.

**Pull request merged**